.github/workflows/deploy_docs.yaml

```diff
@@ -0,0 +1,32 @@
+name: deploy documentation
+on:
+  push:
+    branches:
+      - master
+jobs:
+  hal-cgp-deploy-docs:
+    name: hal-cgp deploy docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: checkout
+        uses: actions/checkout@v2
+      - name: Set up Python 3.8
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+      - name: Install dependencies
+        run: |
+          python3 -m pip install --upgrade pip
```

Comment:
why does this need `python3`? should just `python` work in this env? would keep this more general